testbed/tests/assertions.py

```diff
@@ -28,3 +28,73 @@ def assert_color(actual, expected):
                 expected.b,
                 approx(expected.a, abs=(1 / 255)),
             )
+
+
+def reverse_alpha_blending_over(blended_color, back_color, original_alpha):
+    # This is the reverse of the "over" operation and has
+    # been derived from the "over" operation formula, see:
+    # https://en.wikipedia.org/wiki/Alpha_compositing#Description
+
+    if original_alpha == 0:
+        return back_color
+    else:
+        front_color = rgba(
+            round(
+                (
+                    (blended_color.r * blended_color.a)
+                    - (back_color.r * back_color.a * (1 - original_alpha))
+                )
+                / original_alpha
+            ),
+            round(
+                (
+                    (blended_color.g * blended_color.a)
+                    - (back_color.g * back_color.a * (1 - original_alpha))
+                )
+                / original_alpha
+            ),
+            round(
+                (
+                    (blended_color.b * blended_color.a)
+                    - (back_color.b * back_color.a * (1 - original_alpha))
+                )
+                / original_alpha
+            ),
+            original_alpha,
+        )
+        return front_color
+
+
+def assert_background_color(actual, expected):
+    # For platforms where alpha blending is manually implemented, the
+    # probe.background_color property returns a tuple consisting of:
+    #   - The widget's background color
+    #   - The widget's parent's background color
+    #   - The widget's original alpha value - Required for deblending
+    if isinstance(actual, tuple):
+        actual_widget_bg, actual_parent_bg, actual_widget_bg_alpha = actual
+        deblended_actual_widget_bg = reverse_alpha_blending_over(
+            actual_widget_bg, actual_parent_bg, actual_widget_bg_alpha
+        )
+        if isinstance(expected, tuple):
```

Comment:
How can the *expected* value be a tuple? That's always going to be the reference color, which will be a Toga color literal.

Reply:
There are instances in the test, where 
* `expected` value can be a `tuple`:
```
original = probe.background_color
widget.style.background_color = RED
...
del widget.style.background_color
assert_background_color(probe.background_color, original)
```

* `expected` value can be a single value object:
```
widget.style.background_color = RED
assert_background_color(probe.background_color, named_color(RED))
```

Reply:
Ah - I'd missed the case where the expected color is coming from a probe. In that case - why not just compare the tuples directly? Surely it's a trivial case, same as when both expected and actual are actual colors? Why use the deblending path?

Reply:
We could compare the tuples directly, but we have keep in mind that the probe returned tuple contains blended colors. In cases where the tuple values wouldn't be equal due to different widget color - then it might not be obvious to debug a blended color.
For example:
| |On the Interface|Expected Tuple Value|Probe Returned Tuple Value|
|--|--|--|--|
|Parent Background|`rgba(255,0,0,1)`|`rgba(255,0,0,1)`|`rgba(255,0,0,1)`|
|Widget Background|`rgba(0,0,255,0.5)`|`rgba(127,0,128,1)`|`rgba(0,255,0,1)`|

Now when the `expected widget color value` is compared with the `probe returned widget color value` directly, then it would report the failure to something similar like:
```
assert (actual.r, actual.g, actual.b, actual.a) == (
AssertionError: assert (0,255,0,1) == (127,0,128,1)
```
Then it might not be immediately obvious from where the expected value `rgba(127,0,128,1)` has come. 

Compared this to asserting deblended color, the error message would be something like:
```
assert (actual.r, actual.g, actual.b, actual.a) == (
AssertionError: assert (0,255,0,1) == (0,0,255,0.5)
```

Which clearly specifies that `rgba(0,0,255,0.5)` is the expected color. This would be easy to debug since the expected value would be the same as the one set by the test on the interface.